### ISSUE TITLE
Sensitivity analysis using smoothness restrictions for other periods in staggered treatment timing

### ISSUE THREAD

**Issue by mu2023** [mcaceresb/stata-honestdid#7]
Hi, 
Thank you for writing the Stata package for honest did! 
I am trying to run the sensitivity analysis for other periods in a staggered treatment setting. I use csdid to get my estimates, and run honestdid. I can get he sensitivity results for the default event study (Tp0) time with no issues, and I can also do the sensitivity analysis for other periods using relative magnitudes. For example,  using the example dataset from mixtape, I can get the sensitivity results using relative magnitude restriction by running
`honestdid, pre(3/6) post(8/12) mvec(0.5(0.5)2) coefplot `
The original CIs in the sensitivity table are the same for the relevant relative time produced from csdid.

However, when I try to run the analysis using smoothness restrictions, I can not replicate the analysis for other time periods. I tried playing with l_vec as suggested for the non-staggered case, but it seems like I am missing a point.

Could you please guide me on how to test the sensitivity of other periods using smoothness restrictions? 
Thank you very much.

**Comment by jonathandroth** [comment]
Thank you for your message! To help debug, could you please provide a
minimum working example of the code you ran, the output you expected, and
the output/error that you received. Thanks!

**Comment by jonathandroth** [comment]
Hmm, that is odd. The code that you sent works well for me:

[image: image.png]

Can you please provide your operating system and Stata version? Mauricio
will probably be better-equipped to help you with this than I will.

**Comment by mu2023** [comment]
I am using Stata 16.1 in Windows. 

When I re-ran the code I got a slightly different error message. I can see that it grabs the correct estimate to check the sensitivity for (from original CIs), but no results for Ms:

![image](https://user-images.githubusercontent.com/91473938/201120969-938d6b15-6e43-4450-9c97-3fdb03bec769.png)


Thanks once again.

**Comment by mcaceresb** [comment]
@mu2023 Can you install from the issue branch and re-try?

```stata
net uninstall honestdid
local github https://raw.githubusercontent.com
net install honestdid, from(`github'/mcaceresb/stata-honestdid/issue7) replace
honestdid _plugin_check
```

Should tell you you're using version `1.1.2`; then try again from a fresh Stata session.

---

@jonathandroth You were right that it was a time thing. I missed it when I greped the code quickly in between our meetings, but this is actually documented [here](https://osqp.org/docs/solver/index.html#rho-step-size). I suppose this is done for the sake of computational speed, but it seems to be messing up sometimes.

To be fair to them, it looks to be on average ~10% slower on my machine (without parallel I reckon this could be worse). I'm not too concerned with this, however, since the Stata version was already pretty fast.

In any case, I've pushed a fix to an issue branch (which is to just always adapt rho). If @mu2023 confirms the issue is fixed, can you check `doc/readme.do` and `doc/examples.do` run in your OSX machines? After that I can merge to main.

**Comment by mu2023** [comment]
I re-installed honestdid and now the code works without any warning messages. 
Thanks @mcaceresb and @jonathandroth.

**Comment by jonathandroth** [comment]
Mauricio -- I've run this on both Intel and M1 silicon without a problem,
so I think we should be good to merge.

**Comment by mcaceresb** [comment]
Fix merged in #8; see [here](https://github.com/mcaceresb/stata-honestdid/pull/8#issue-1448603337) for details.
